packages/story-editor/src/components/panels/shared/hotlinkModal.js

```diff
@@ -0,0 +1,205 @@
+/*
+ * Copyright 2022 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * External dependencies
+ */
+import { __, sprintf, translateToExclusiveList } from '@googleforcreators/i18n';
+import { Input } from '@googleforcreators/design-system';
+import PropTypes from 'prop-types';
+import styled from 'styled-components';
+import {
+  useCallback,
+  useRef,
+  useState,
+  useLayoutEffect,
+} from '@googleforcreators/react';
+import { trackError, trackEvent } from '@googleforcreators/tracking';
+import { withProtocol } from '@googleforcreators/url';
+/**
+ * Internal dependencies
+ */
+import Dialog from '../../dialog';
+import {
+  isValidUrlForHotlinking,
+  getErrorMessage,
+} from '../../library/panes/media/local/hotlink/utils';
+import { useAPI } from '../../../app/api';
+import useCORSProxy from '../../../utils/useCORSProxy';
+
+const InputWrapper = styled.form`
+  margin: 16px 4px;
+  width: 470px;
+  height: 100px;
+`;
+
+function HotlinkModal({
+  isOpen,
+  onClose,
+  onSelect,
+  allowedFileTypes = [],
+  title,
+}) {
+  const [isInserting, setIsInserting] = useState(false);
+  const [link, setLink] = useState('');
+  const [errorMsg, setErrorMsg] = useState(false);
+  const inputRef = useRef(null);
+
+  const {
+    actions: { getHotlinkInfo },
+  } = useAPI();
+
+  const { checkResourceAccess } = useCORSProxy();
+
+  const isDisabled = errorMsg || !link || isInserting;
+  const primaryText = isInserting
+    ? __('Inserting…', 'web-stories')
+    : __('Insert', 'web-stories');
+
+  let description = __('No file types are currently supported.', 'web-stories');
+  if (allowedFileTypes.length) {
+    description = sprintf(
+      /* translators: %s is a list of allowed file extensions. */
+      __('You can insert %s.', 'web-stories'),
```

Comment:
Should this be a little more descriptive ?

Current `You can insert vtt.`

Maybe - You can insert a .vtt file

Reply:
Agreed.

I just flagged this in #10515 to get some input from Amy & Andrew.